applications/matter_bridge/src/bridged_device_types/onoff_light_data_provider.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "onoff_light_data_provider.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace ::chip;
+using namespace ::chip::app;
+
+void OnOffLightDataProvider::Init()
+{
+	k_timer_init(&mTimer, OnOffLightDataProvider::TimerTimeoutCallback, nullptr);
+	k_timer_user_data_set(&mTimer, this);
+	k_timer_start(&mTimer, K_MSEC(kOnOffIntervalMs), K_MSEC(kOnOffIntervalMs));
+}
+
+void OnOffLightDataProvider::NotifyUpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId, void *data,
+					       size_t dataSize)
+{
+	if (mUpdateAttributeCallback) {
+		mUpdateAttributeCallback(*this, Clusters::OnOff::Id, Clusters::OnOff::Attributes::OnOff::Id, data,
+					 dataSize);
+	}
+}
+
+CHIP_ERROR OnOffLightDataProvider::UpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId,
+					       uint8_t *buffer)
+{
+	if (clusterId != Clusters::OnOff::Id) {
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+
+	switch (attributeId) {
+	case Clusters::OnOff::Attributes::OnOff::Id: {
+		mOnOff = *buffer;
+		NotifyUpdateState(clusterId, attributeId, &mOnOff, sizeof(mOnOff));
```

Comment:
Not a blocker for this PR but should we really call `NotifyUpdateState()` here? To me, the the current flow is as follows:
- the Matter attribute is altered
- we call `BridgeManager::HandleWrite` which results in updating the bridged device state AND synchronizing the data provider as well by issuing polymorphic `BridgedDeviceDataProvider::UpdateState`
- in `OnOffLightDataProvider::UpdateState` we call `NotifyUpdateState` which brings us back to `BridgeManager::HandleUpdate` resulting in Matter bridged device attribute update which essentially was a trigger of the change in the first place

Reply:
Do you suggest repairing it here or unblocking your PR and considering it as a follow-up?

Reply:
I think we need @kkasperczyk-no 's insight here. For me we can live with it for now and fix later if needed.

Reply:
Ok, leave it as it is currently, and make a follow-up since it is the base of the whole project. :)